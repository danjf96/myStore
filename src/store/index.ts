
import { applyMiddleware, combineReducers, createStore } from 'redux'
import products from './ducks/products'
import shoppingCart from './ducks/shoppingCart'
import ReduxThunk from 'redux-thunk';

const reducers = {
    products,
    shoppingCart
}

const combinedReducers =  combineReducers(reducers)

const store = createStore(combinedReducers, {}, applyMiddleware(ReduxThunk));

// We can use RootState type in every file in project
declare global {
    type RootState = ReturnType<typeof store.getState>
}

// Thanks to that you will have ability to use useSelector hook with state value
declare module 'react-redux' {
    interface DefaultRootState extends RootState { }
}

export default store

