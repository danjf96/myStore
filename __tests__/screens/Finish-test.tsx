/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Finish from '../../src/screens/finish';
 import { render } from 'react-native-testing-library';
import { Provider } from 'react-redux';
import store from '../../src/store';
 const component = (
  <Provider store={store} > <Finish navigation={{ navigate: (screen:'string') => null }}/> </Provider>
);
 it('renders correctly', async () => {
   const screen = render(component)
   expect(screen.toJSON()).toMatchSnapshot()
 });
  