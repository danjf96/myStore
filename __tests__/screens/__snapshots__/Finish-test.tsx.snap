// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0086ff",
        "flexDirection": "column",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          exclusive={true}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "marginBottom": 40,
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰅖
            </Text>
          </View>
        </RNGestureHandlerButton>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 25,
              "fontWeight": "bold",
              "marginBottom": 40,
            }
          }
        >
          Compra finalizada com sucesso!
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "marginBottom": 20,
            }
          }
        >
          Valor total de R$ 
          0,00
          . Saldo atual de R$ 585,00
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 60,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󱊛
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "marginRight": 10,
              }
            }
          >
            Sua compra chegará em até 3 dias úteis
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>,
  " ",
]
`;
